lib/Collection.php

```diff
@@ -73,6 +73,7 @@ public function __construct($uri, $graph)
      * @throws \OutOfBoundsException
      * @throws \InvalidArgumentException
      */
+    #[\ReturnTypeWillChange]
```

Comment:
This annotation allows us to suppress a deprecation notice triggered by PHP 8.1 - it will require fixing once PHP 9 is released. Since PHP 7.1 is the minimum version we could choose to add the return types here. I went the annotation route because changing or adding a return type should have a new major as per semver.